Update R to 3.5.1

JanJecmen opened the pull request.

JanJecmen committed:
move vanilla R to 3.5.1

JanJecmen committed:
switch to patched R 3.5.1

JanJecmen committed:
Build cleanup

* simplify cmakelists
* switch to c++17
* fix svn revision for building gnur
* remove some unused stuff

JanJecmen committed:
Migrate RIR for R 3.5.1

* the compiler api changed and now the compiler is supposed to modify the closure inplace
* removed unused environment flags
* sexp header changed to be 64 bits, changed the code magic to reflect this

JanJecmen committed:
fix rir tests

JanJecmen committed:
try macos build

mhyee commented on CMakeLists.txt:
```diff
@@ -11,11 +11,11 @@ set(R_INCLUDE_DIR ${R_HOME}/include)
-set(CMAKE_C_FLAGS_DEBUG "-Og -std=gnu99 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS_RELEASE "-O2")
+set(CMAKE_CXX_FLAGS_DEBUG "-O0 -DENABLE_SLOWASSERT")
+set(CMAKE_CXX_FLAGS "-Wall -std=c++17 -fno-rtti")
```
Well, if we're skipping R 3.4, might as well skip C++14!

I look forward to ~spamming~ using:

- generic lambdas
- s suffix for `std::string`
- `static_assert` without the message string
- nested namespace definitions
- static `if`
- `std::optional`

mhyee reviewed (commented):
Another thing to update is the Travis config. It's set to Trusty (14.04!), so of course GCC is version 4.8.4 and doesn't understand `-std=c++17`. The latest Ubuntu LTS is Bionic Beaver (18.04).

JanJecmen committed:
Update .travis.yml

JanJecmen committed:
Update .travis.yml

JanJecmen committed:
Update .travis.yml

JanJecmen committed:
try setting empty tz

JanJecmen committed:
compiler on macos

mhyee committed:
Try fixing tz issue on macOS build (again)

- Change the config to use `--with-internal-tzcode` on macOS
    - This is supposed to be the default, because the system
      implementation of time zones is broken
    - Not sure why we changed this, but macOS and R have both changed
      since then

mhyee committed:
Make `--with-internal-tzcode` explicit

...rather than relying on the default setting

mhyee committed:
Try removing TZ environment variable

The test in question (datetime.R) manually overrides the value of TZ.

mhyee committed:
Reverse previous commit

Some other test needs TZ environment variable to be empty

JanJecmen commented:
Oh you fixed it! Thanks!!

JanJecmen commented:
TBR then, I guess :)

o- commented:
Yet another :tada:
